Tests/NIOTests/StreamChannelsTest.swift

```diff
@@ -246,6 +247,85 @@ class StreamChannelTest: XCTestCase {
+        class FailOnReadHandler: ChannelInboundHandler {
+            typealias InboundIn = ByteBuffer
+
+            let areReadOkayNow: NIOAtomic<Bool>
+
+            init(areReadOkayNow: NIOAtomic<Bool>) {
+                self.areReadOkayNow = areReadOkayNow
+            }
+
+            func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+                guard self.areReadOkayNow.load() else {
+                    XCTFail("unexpected read of \(self.unwrapInboundIn(data))")
+                    return
+                }
+            }
+
+            func channelReadComplete(context: ChannelHandlerContext) {
+                guard self.areReadOkayNow.load() else {
+                    XCTFail("unexpected readComplete")
+                    return
+                }
+            }
+        }
+
+        func runTest(receiver: Channel, sender: Channel) throws {
+            let sends = NIOAtomic<Int>.makeAtomic(value: 0)
+            precondition(receiver.eventLoop !== sender.eventLoop,
+                         "this test cannot run if sender and receiver live on the same EventLoop. \(receiver)")
+            XCTAssertNoThrow(try receiver.setOption(ChannelOptions.autoRead, value: false).wait())
+            let areReadOkayNow: NIOAtomic<Bool> = .makeAtomic(value: false)
+            XCTAssertNoThrow(try receiver.pipeline.addHandler(FailOnReadHandler(areReadOkayNow: areReadOkayNow)).wait())
+
+            // We will immediately send exactly the amount of data that fits in the receiver's receive buffer.
+            let receiveBufferSize = Int((try? receiver.getOption(ChannelOptions.socket(.init(SOL_SOCKET),
+                                                                                       .init(SO_RCVBUF))).wait()) ?? 8192)
+            var buffer = sender.allocator.buffer(capacity: receiveBufferSize)
+            buffer.writeBytes(Array(repeating: UInt8(ascii: "X"), count: receiveBufferSize))
+
+            XCTAssertNoThrow(try sender.eventLoop.submit {
+                func send() {
+                    var allBuffer = buffer
+                    // When we run through this for the first time, we send exactly the receive buffer size, after that
+                    // we send one byte at a time. Sending the receive buffer will trigger the EVFILT_EXCEPT loop
+                    // (rdar://53656794) for UNIX Domain Sockets and the additional 1 byte send loop will also pretty
+                    // reliably trigger it for TCP sockets.
+                    let myBuffer = allBuffer.readSlice(length: sends.load() == 0 ? receiveBufferSize : 1)!
+                    sender.writeAndFlush(myBuffer).map {
+                        _ = sends.add(1)
+                        sender.eventLoop.scheduleTask(in: .microseconds(1)) {
+                            send()
+                        }
+                    }.whenFailure { error in
+                        guard case .some(.ioOnClosedChannel) = error as? ChannelError else {
```

Comment:
nit: I think it is kind of odd to use a guard here if there is nothing defined outside of the scope of the guard itself. So maybe just use a `if let` ?

Reply:
discussed offline, no really good way to spell with `if` because there's no `if not case`